lib/ansible/modules/cloud/amazon/ec2_ami_facts.py

```diff
@@ -0,0 +1,229 @@
+      - See U(https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeImages.html) for possible filters.
+      - Filter names and values are case sensitive.
+  owners:
+    description:
+      - Filter the images by the owner. Valid options are an AWS account ID, self,
+      - or an AWS owner alias ( amazon | aws-marketplace | microsoft ).
+    aliases: [owner]
+  executable_users:
+    description:
+      - Filter images by users with explicit launch permissions. Valid options are an AWS account ID, self, or all (public AMIs).
+    aliases: [executable_user]
+
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: gather facts about an AMI using ami-id
+  ec2_ami_facts:
+    image_ids: ami-5b488823
+
+- name: gather facts about all AMIs with tag key Name and value webapp
+  ec2_ami_facts:
+    filters:
+      "tag:Name": webapp
+
+- name: gather facts about an AMI with 'AMI Name' equal to foobar
+  ec2_ami_facts:
+    filters:
+      name: foobar
+
+- name: gather facts about Ubuntu 17.04 AMIs published by Canonical (099720109477)
+  ec2_ami_facts:
+    owners: 099720109477
+    filters:
+      name: "ubuntu/images/ubuntu-zesty-17.04-*"
+'''
+
+RETURN = '''
+images:
+  description: a list of images
+  returned: always
+  type: complex
+  contains:
+    architecture:
+      description: The architecture of the image
+      returned: always
+      type: string
+      sample: x86_64
+    block_device_mappings:
+      description: Any block device mapping entries
+      returned: always
+      type: complex
+      contains:
+        device_name:
+          description: The device name exposed to the instance
+          returned: always
+          type: string
+          sample: /dev/sda1
+        ebs:
+          description: EBS volumes
+          returned: always
+          type: complex
+    creation_date:
+      description: The date and time the image was created
+      returned: always
+      type: string
+      sample: '2017-10-16T19:22:13.000Z'
+    description:
+      description: The description of the AMI
+      returned: always
+      type: string
+      sample: ''
+    ena_support:
+      description: whether enhanced networking with ENA is enabled
+      returned: always
+      type: bool
+      sample: true
+    hypervisor:
+      description: The hypervisor type of the image
+      returned: always
+      type: string
+      sample: xen
+    image_id:
+      description: The ID of the AMI
+      returned: always
+      type: string
+      sample: ami-5b466623
+    image_location:
+      description: The location of the AMI
+      returned: always
+      type: string
+      sample: 408466080000/Webapp
+    image_type:
+      description: The type of image
+      returned: always
+      type: string
+      sample: machine
+    name:
```

Comment:
It's worth returning the launch permissions (you need to run `describe_image_attribute` separately to get this)

Reply:
@willthames - Thanks. I have made the requested changes.